lib/orderbook/OrderBook.ts

```diff
@@ -279,19 +268,19 @@ class OrderBook extends EventEmitter {
         this.emit('ownOrder.filled', portion);
         onUpdate && onUpdate({ case: PlaceOrderEventCase.InternalMatch, payload: maker });
       } else {
-        if (!this.swaps || !this.swaps.verifyExecution(maker, taker)) {
-          rejectNonInternalMatch(taker, maker);
+        if (!this.swaps) {
           continue;
         }
 
-        this.emit('peerOrder.filled', portion); // make sure we emit this event on every case in which we don't re-add the maker order
         try {
           const swapResult = await this.swaps.executeSwap(maker, taker);
+          this.emit('peerOrder.filled', portion);
           result.swapResults.push(swapResult);
           onUpdate && onUpdate({ case: PlaceOrderEventCase.SwapResult, payload: swapResult });
         } catch (err) {
-          // we can either push to swapFailures, or reject in case of non-retry errors
+          this.emit('peerOrder.invalidation', portion);
```

Comment:
Although it is nice to have for testing you might want to put in a todo so that we don't forget to remove this line in the future.

Reply:
Remove the `this.emit('peerOrder.invalidation', portion);`? In my mind this is deliberate, we want to get rid of peer orders that we can't swap.